Add validation for KUBE_USER

adityakali opened the pull request:
Malformed KUBE_USER causes error in cluster setup.

cc/ @kubernetes/goog-image 
@Q-Lee @Amey-D  Can you please review?

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/31318)

<!-- Reviewable:end -->

adityakali committed:
Add validation for KUBE_USER

Malformed KUBE_USER causes error in cluster setup.

adityakali commented:
This is needed in next 1.3.x cut as well. Can someone please set appropriate labels for it?
Thanks.

rmmh commented on cluster/gce/gci/configure-helper.sh:
```diff
@@ -752,10 +752,10 @@ function start-kube-apiserver {
   local -r src_dir="${KUBE_HOME}/kube-manifests/kubernetes/gci-trusty"
 
-  if [[ -n "${KUBE_USER:-}" ]]; then
+  if [[ -n "${KUBE_USER}" ]]; then
```
This should still be `"${KUBE_USER:-}"` if this script is supposed to work without KUBE_USER being set.

adityakali replied:
KUBE_USER is now initialized early on in the setup (on line 1119) if its not set. So its redundant to do it here.

rmmh replied:
No, `"${KUBE_USER:-}"` doesn't set KUBE_USER. Use `${KUBE_USER:=admin}` if you want to initialize it.

Q-Lee replied:
Please fix this under trusty as well.

adityakali replied:
> Please fix this under trusty as well.

The change in cluster/common.sh takes care of trusty too, no?

adityakali replied:
> No, ${KUBE_USER:-} doesn't set KUBE_USER. Use ${KUBE_USER:=admin} if you want to initialize it.

Ah. I assumed it also sets it. I restored it as before. But I think with `set -o nounset` at the top of this file, even `${KUBE_USER:-}` will cause this code to fail if KUBE_USER is unset. I also see KUBE_USER used with the not-null check above.

spxtr replied:
With nounset, `${THIS_IS_UNSET:-}` will not fail. [This is a super-useful resource](http://tldp.org/LDP/abs/html/parameter-substitution.html).

adityakali committed:
Validate KUBE_USER and KUBE_PASSWORD early on

Make sure KUBE_USER and KUBE_PASSWORD contains valid chars
as soon as we read them from kube config.

adityakali commented:
Updated the PR with additional checks. PTAL.
